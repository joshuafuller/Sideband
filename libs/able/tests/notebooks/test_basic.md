---
jupyter:
  jupytext:
    formats: ipynb,md
    text_representation:
      extension: .md
      format_name: markdown
      format_version: '1.3'
      jupytext_version: 1.11.2
  kernelspec:
    display_name: Python 3
    language: python
    name: python3
---

# Setup

```python
%run init.ipynb
```

```python
%%there
class BLE(BluetoothDispatcher):

    def on_scan_started(self, success):
        results.started = success
        
    def on_scan_completed(self):
        results.completed = 1
        
    def on_device(self, device, rssi, advertisement):
        results.devices.append(device)
        
ble = BLE()
```

# Run BLE devices scan

```python
%%there
results =  Results()
print(f"Started: {results.started} Completed: {results.completed}")
ble.start_scan()
```

```python
sleep(10)
```

```python
%%there
ble.stop_scan()
```

```python
sleep(2)
```

# Check that scan started and completed

```python
%%there
print(f"Started: {results.started} Completed: {results.completed}")
```

# Check that testing device was discovered

```python
%%there
print(
    "KivyBLETest" in [dev.getName() for dev in results.devices]
)
```
