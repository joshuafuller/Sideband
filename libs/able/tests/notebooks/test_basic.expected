[[setup]]
= Setup

[[run-ble-devices-scan]]
= Run BLE devices scan


----
Started: None Completed: None
----

[[check-that-scan-started-and-completed]]
= Check that scan started and completed


----
Started: 1 Completed: 1
----

[[check-that-testing-device-was-discovered]]
= Check that testing device was discovered


----
True
----
